docs/faq/code-analysis/how-long-does-it-take-for-my-repository-to-be-analyzed.md

```diff
@@ -1,6 +1,13 @@
 # How long does it take for my repository to be analyzed?
 
-Codacy usually takes under 5 minutes to analyze your repository, however it may take longer as this depends on the size of your repository.
+Codacy usually takes under 5 minutes to analyze your repository. It may however take longer, depending on a number of factors:
+
+-   The size of your repository
+-   Whether it's the initial analysis of your repository
+-   The priority of your analysis request, with open source projects having lower priority
+-   The current load on Codacy's servers
+-   The time it takes your Git provider to call the webhooks on Codacy
+-   Whether Codacy is currently experiencing issues or outages
```

Comment:
Maybe we should also consider issues or outages of the cloud Git provider?

Reply:
One question this raises is that we have two classes of issues: those the users can do something about (e.g. repo size, by ignoring some files perhaps), and those that the users have to endure (e.g. outages or overall load).

Should we separate the two classes somehow?

Reply:
I'd mention issues both on the Codacy side and the Git provider side in the same sentence, especially since there's not much else that users can do in these situations:

```suggestion
-   Whether Codacy or your Git provider is currently experiencing issues or outages
```

Reply:
> Should we separate the two classes somehow?

This seems like a good idea, yes. :+1: 

We could also try ordering the items in these classes in some logical order, for example, chronological order.